acct/chsh

```diff
@@ -1,38 +1,37 @@
-#!/usr/bin/env python
-from __future__ import print_function
-
+#!/usr/bin/env python3
 import getpass
-import os
+import subprocess
 import sys
 
-import ldap
-import ldap.modlist as modlist
-import ldap.sasl as sasl
+from ocflib.account.manage import modify_ldap_attributes
+from ocflib.misc.validators import VALID_LOGIN_SHELLS
 
 if len(sys.argv) < 2:
-    print('Usage: chsh shell')
-    sys.exit()
-
-shell = sys.argv[1]
-etcshells = open('/etc/shells', 'r')
-line = etcshells.readline()
-while line != '':
-    validshell = line.strip()
-    if shell == validshell:
-        print('Changing shell to %s' % (validshell))
-        kinit = '/usr/bin/klist -t || /usr/bin/kinit'
-        os.system(kinit)
-        conn = ldap.initialize('ldaps://ldap.ocf.berkeley.edu:636/')
-        auth = sasl.gssapi('')
-        conn.sasl_interactive_bind_s('', auth)
-        dn = 'uid=%s,ou=people,dc=ocf,dc=berkeley,dc=edu' % (getpass.getuser())
-        old = {'loginShell': '*'}
-        new = {'loginShell': '%s' % (validshell)}
-        ldif = modlist.modifyModlist(old, new)
-        conn.modify_s(dn, ldif)
-        conn.unbind_s()
-        print('Shell successfully changed, typically it takes a while for the '
-              'changes to propagate to all our machines')
-        sys.exit()
-    line = etcshells.readline()
-print('The shell you chose is not valid, please chose another shell')
+    print('Usage: chsh [shell]')
+    sys.exit(1)
+
+new_shell = sys.argv[1]
+
+print('Changing shell to {}...'.format(new_shell))
+
+# Check if the user has a valid kerberos ticket
+has_kerberos = subprocess.run(('/usr/bin/klist', '-t')).returncode
```

Comment:
perhaps use `subprocess.check_call`?

Reply:
I don't see a reason to. Also the Python docs say this about run:
>The recommended approach to invoking subprocesses is to use the run() function for all use cases it can handle.

Reply:
> check_call() and check_output() will raise CalledProcessError if the called process returns a non-zero return code.

(from https://docs.python.org/3.5/library/subprocess.html#exceptions)

Given we don't want an exception here and just want the return code, I think using `subprocess.run` here is better.